modules/simple_bucket/variables.tf

```
@@ -150,3 +150,11 @@ variable "public_access_prevention" {
   type        = string
   default     = "inherited"
 }
+
+variable "soft_delete_policy" {
+  description = "Soft delete policies to apply. Format is the same as described in provider documentation https://www.terraform.io/docs/providers/google/r/storage_bucket.html#nested_soft_delete_policy"
+  type = object({
+    retention_duration_seconds = number
+  })
+  default = { retention_duration_seconds = 604800 }
```

So we can default to the provider default

```suggestion
  default = { }
```

---

Okay, made it optional for the type to work